src/main/scala/org/hammerlab/guacamole/assembly/DeBruijnGraph.scala

```diff
@@ -354,17 +367,103 @@ object DeBruijnGraph {
   }
 
   /**
-   * Merge sequences where we expect consecutive entries to overlap by `overlapSize` bases.
-   *
-   * @param sequences Set of sequences that we would like to combine into a single sequence
-   * @param overlapSize The amount of the sequences we expect to overlap (The number of bases the last sequence overlaps
-   *                    with the next. For a standard kmer graph, this is the length of the kmer length - 1
-   * @return A single merged sequence
-   */
+    * Merge sequences where we expect consecutive entries to overlap by `overlapSize` bases.
+    *
+    * @param sequences Set of sequences that we would like to combine into a single sequence
+    * @param overlapSize The amount of the sequences we expect to overlap (The number of bases the last sequence overlaps
+    *                    with the next. For a standard kmer graph, this is the length of the kmer length - 1
+    *
+    * @return A single merged sequence
+    */
   def mergeOverlappingSequences(sequences: Seq[Sequence], overlapSize: Int): Sequence = {
     val head = sequences.headOption.getOrElse(Seq.empty)
     val rest = sequences.tail.flatMap(sequence => sequence.takeRight(sequence.length - overlapSize + 1))
     head ++ rest
   }
 
+  /**
+   * Extract a subsequence from a read sequence
+   *
+   * NOTE: This assumes that the read entirely covers the reference region and does not check for insertions
+   * or deletion
```

Comment:
`s/deletion/deletions/`.

Can you expound in the comment on why it's safe to assume no insertions or deletions here?

This feels like it should be named something like `sliceRead` or even made a method on `MappedRead` called `slice`. `getSequenceFromRead` sounds like it would just return `read.sequence`.